tests/tests/bitasset_tests.cpp

```diff
@@ -542,6 +544,455 @@ BOOST_AUTO_TEST_CASE( hf_890_test )
+      creator.common_options.max_supply = 0;
+      creator.precision = 6;
+      creator.common_options.core_exchange_rate = price({asset(1,asset_id_type(1)),asset(1)});
+      creator.common_options.max_supply = GRAPHENE_MAX_SHARE_SUPPLY;
+      creator.common_options.flags = charge_market_fee;
+      creator.common_options.issuer_permissions = charge_market_fee;
+      trx.operations.push_back(std::move(creator));
+      trx.validate();
+      processed_transaction ptx = db.push_transaction(trx, ~0);
+      trx.operations.clear();
+      const asset_object& six = db.get<asset_object>(ptx.operation_results[0].get<object_id_type>());
+      sixbit = &six;
+   }
+   catch( fc::exception& ex )
+   {
+      BOOST_FAIL(ex.to_string(log_level_all));
+   }
+
+   BOOST_TEST_MESSAGE( "Create Prediction market coin with precision of 6, backed by SIXBIT" );
+   const asset_object* prediction = nullptr;
+   try
+   {
+      asset_create_operation creator;
+      creator.issuer = GRAPHENE_WITNESS_ACCOUNT;
+      creator.fee = asset();
+      creator.symbol = "PREDICTION";
+      creator.common_options.max_supply = GRAPHENE_MAX_SHARE_SUPPLY;
+      creator.precision = 6;
+      creator.common_options.market_fee_percent = 100;
+      creator.common_options.issuer_permissions = charge_market_fee | global_settle;
+      creator.common_options.flags = charge_market_fee & ~global_settle;
+      creator.common_options.flags |= witness_fed_asset;
+      creator.common_options.core_exchange_rate = price({asset(1,asset_id_type(1)),asset(1)});
+      creator.bitasset_opts = bitasset_options();
+      creator.bitasset_opts->short_backing_asset = sixbit->get_id();
+      creator.is_prediction_market = true;
+      trx.operations.push_back(std::move(creator));
+      trx.validate();
+      processed_transaction ptx = db.push_transaction(trx, ~0);
+      trx.operations.clear();
+      const asset_object& pred = db.get<asset_object>(ptx.operation_results[0].get<object_id_type>());
+      prediction = &pred;
+   }
+   catch (fc::exception& ex)
+   {
+      BOOST_FAIL(ex.to_string(log_level_all));
+   }
+
+   // make a generic operation
+   bitasset_evaluator_wrapper evaluator;
+   evaluator.set_db(db);
+   asset_update_bitasset_operation op;
+   op.asset_to_update = bit_usd_id;
+   op.issuer = bit_usd.issuer;
+   op.new_options = bit_usd.bitasset_data(db).options;
+
+   // this should pass
+   BOOST_TEST_MESSAGE("Evaluating a good operation");
+   BOOST_CHECK( evaluator.evaluate(op) == void_result() );
+
+   // test with a market issued asset
+   BOOST_TEST_MESSAGE( "Sending a non-bitasset." );
+   op.asset_to_update = blah.get_id();
+   try
+   {
+      evaluator.evaluate(op);
+      BOOST_FAIL("Sending a non-bitasset should have failed.");
+   }
+   catch (fc::exception& ex)
+   {
+      BOOST_CHECK_EQUAL( ex.to_string(log_level_all), "Assert Exception: asset_obj.is_market_issued(): Cannot update BitAsset-specific settings on a non-BitAsset." );
```

Comment:
Hm, this construct is much more complicated than a simple BOOST_CHECK_THROW or GRAPHENE_REQUIRE_THROW.
Since you're using it repeatedly you might want to add a macro to database_fixture.hpp.

Generally I think it's a good idea to check that the correct assertion failed. But an exact exception string comparison has the downside that a simple textual change will make the test fail, which means the text must be maintained in two places. Perhaps use an inexact comparison (substring or regex).

Reply:
Added macro to database_fixture.hpp that looks for some text in exception.what()